src/config/app.spec.ts

```diff
@@ -0,0 +1,32 @@
+import {PathLike} from 'fs';
+import createConfig from './app';
+
+describe('config/app', () => {
+    describe('createConfig()', () => {
+        const phpIniFromConfigurationPath: PathLike = 'tests/php-ini-from-configuration';
+
+        it('should return valid config', () => {
+            expect(createConfig(
+                {
+                    codeChecks : true,
+                    docLinting : true,
+                },
+                `${phpIniFromConfigurationPath}/composer.json`,
+                `${phpIniFromConfigurationPath}/composer.lock`,
+                `${phpIniFromConfigurationPath}/.laminas-ci.json`
+            )).toEqual({
+                codeChecks                    : true,
+                docLinting                    : true,
+                versions                      : [ '8.1' ],
```

Comment:
One thing I realized: `~8.1.2` will, for example, never match `8.1` 🤔

Reply:
It's normalizing all the versions to these minor version numbers beforehand, so it should in theory (see: https://github.com/laminas/laminas-ci-matrix-action/blob/d9110cbc257d7d10cdd4b5345256386c9024c632/src/config/app.ts#L31).

Reply:
Do we have an existing test for that scenario?

Reply:
Good question, I'm not even sure what's supposed to be returned here, I'm taking the code at face value. Could do with waiting for @boesing (time permitting obviously) to have a look over this, and I can add more tests after that.

Reply:
I associated many PRs that are affected by this specific problem: they can be seen on the PR overview.

The problem is that a dependency range like `"php": "~8.1.12"` will produce a matrix containing PHP 7.4, 8.0, 8.1, for some reason.

Reply:
Ohhhhhhhhhhhhhhhhh, I see the problem now ... It's that line above, if `composerPhpVersion` is `~8.1.12` then `${version}.0` won't match it. I think that's the issue here.

Reply:
It was the line mentioned above after all. `composerPhpVersion` is the constraint pulled straight from `require.php` in `composer.json`.